src/zos.cc

```diff
@@ -3157,6 +3157,55 @@ extern "C" int chdir_long(char *dir) {
   return chdir(dir);
 }
 
+extern "C" const char* getprogname() {
+  char argv[PATH_MAX];
+  W_PSPROC buf;
+  int token = 0;
+  pid_t mypid = getpid();
+
+  memset(&buf, 0, sizeof(buf));
+  buf.ps_pathlen = PATH_MAX;
+  buf.ps_pathptr = &argv[0];
+
+  while ((token = w_getpsent(token, &buf, sizeof(buf))) > 0) {
```

Comment:
this on first blush looks different than what is in the links above. Can you confirm it is consistent and also that the header we need to ensure we get the ASCII variant is present?

Reply:
Functionally, it should be the same besides returning "?" if nothing is found. https://www.gnu.org/software/gnulib/manual/html_node/progname-and-getprogname.html 

However, the getprogname spec doesn't mention anything about returning ?. https://man.freebsd.org/cgi/man.cgi?query=getprogname&sektion=3  
> The getprogname() function returns the name of  the  program.   If  the
       name has	not been set yet, it will return NULL.

The gnulib implementation is slightly different, though. They seem to have copied the example here: https://www.ibm.com/docs/en/zos/2.1.0?topic=functions-w-getpsent-get-process-data

Yep, ps.h is included in https://github.com/ibmruntimes/zoslib/blob/zopen/src/zos.cc#L40.